sr_ur_robot_hw/src/sr_ur_robot_state_client.cpp

```diff
@@ -126,7 +152,7 @@ static void robot_state_received_cb(uv_stream_t* state_stream,
     }
   }
 
-  ur_robot_state *robot_state = reinterpret_cast<ur_robot_state*>(pdata);
+  ur_robot_state_v3_5 *robot_state = reinterpret_cast<ur_robot_state_v3_5*>(pdata);
```

Comment:
@toliver do we need an if here to have either ur_robot_state_v3_2 or ur_robot_state_v3_5?

Reply:
Not really, as we never access the last fields of these structures.
That's also the reason why we use the size of the longest structure to allocate memory for the buffer.

Reply:
I agree this should be refactored to have a nicer way of coping with the different structures.
Actually we might be OK with simply assuming everything is cast to a structure with the latest version.
The reason is that for the moment all the data fields we care about are common to all versions. The new data fields that are added in each subsequent version are not used in this driver.
Although maybe at some point we might want to use them or publish them, so maybe we should distinguish versions.